### PR TITLE
preload: continue running CI job if preloading fails

### PR DESCRIPTION
**Pull request by mike-sul**
Signed-off-by: Mike Sul <mike.sul@foundries.io>

### CODE REVIEW

**Comment by mike-sul** [comment]
`|  INFO: Apps Fetcher: dockerd: Containerd daemon has been stopped
|  tar: can't change directory to '/non-exist': No such file or directory
|  ERROR: Apps Fetcher: apps_fetcher: Failed to fetch Target apps and images: Command '['tar', '-cf', '/var/cache/bitbake/app-images/10931d1969d45586c112dd5105524f9551ede3df/apps-10931d1969d45586c112dd5105524f9551ede3df-amd64.tar', '-C', '/non-exist', '.']' returned non-zero exit status 1.
|  Traceback (most recent call last):
|    File "/ci-scripts/apps/apps_fetcher.py", line 44, in <module>
|      store.store(target, apps_fetcher.target_dir(target.name))
|    File "/ci-scripts/apps/target_apps_store.py", line 44, in store
|      subprocess.check_call(['tar', '-cf', app_image_tar, '-C', images_dir, '.'])
|    File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
|      raise CalledProcessError(retcode, cmd)
|  subprocess.CalledProcessError: Command '['tar', '-cf', '/var/cache/bitbake/app-images/10931d1969d45586c112dd5105524f9551ede3df/apps-10931d1969d45586c112dd5105524f9551ede3df-amd64.tar', '-C', '/non-exist', '.']' returned non-zero exit status 1.
|  
|--
Signing local TUF targets
== 2021-03-29 09:09:59 Running: garage-sign targets sign --repo /tuf-repo/msul-dev01 --key-name targets
|  signed targets.json to /tuf-repo/msul-dev01/roles/targets.json
|--
Your apps has been successfully published, see the work dir for details: /tmp/publish-compose-app-5n1vC005fL
`

File: apps/apps_fetcher.py [hunk 1]
```diff
@@ -42,7 +43,9 @@ def get_args():
         for target, _ in apps_fetcher.target_apps.items():
             store.store(target, apps_fetcher.target_dir(target.name))
     except Exception as exc:
-        logging.error('Failed to fetch Target apps and images: {}'.format(exc))
-        exit_code = 1
+        # since preloading/fetching of App container images are not really mandatory
+        # we return successful exit code and just output an error message along with s stack trace
+        logging.error('Failed to fetch Target apps and images: {}\n{}'.format(exc, traceback.format_exc()))
```
Comment (doanac) [inline apps/apps_fetcher.py:48]: can we do a `logging.exception` to capture the actual stack trace. i've found myself having to make this change and then re-run ci jobs in the past to see where the actual bug is coming from.

Comment (mike-sul) [reply apps/apps_fetcher.py:48]: what's wrong with `traceback.format_exc`, as you can see in the example I provided above it prints the stack trace?

Comment (doanac) [reply apps/apps_fetcher.py:48]: didn't read close enough.

**Review by doanac — APPROVED** [review]

**Pull request merged**